src/k-provider/ovp/provider-parser.js

```diff
@@ -287,14 +287,9 @@ export default class ProviderParser {
    * @private
    */
   static _getBaseProtocol(): string {
-    let splittedUrl: Array<string> = config.baseUrl.split("/");
-    let baseProtocol: string;
-    if (splittedUrl && splittedUrl.length > 0) {
-      baseProtocol = splittedUrl[0].substring(0, splittedUrl[0].length - 1);
-    }
-    else {
-      baseProtocol = "http";
-    }
-    return baseProtocol;
+    const protocolRegex = /^https?:/;
+    const result = protocolRegex.exec(config.baseUrl);
+    const protocol = result ? result[0] : document.location.protocol;
```

Comment:
need to validate `document.location.protocol` as well, as you can't be sure it will be a string, and then the next line with `split` will explode